apps/client/sttr_build/index.js

```diff
@@ -77,31 +77,37 @@ function checkStatus(res) {
       },
     })
       .then(checkStatus)
-      .then((res) => res.text())
-      .then((xml) => ({
-        id,
-        xml,
-      }))
+      .then((res) => res.json())
+      .then((json) => {
+        fs.writeFile(
+          path.join(outputDir, `${id}.source.json`),
+          jsonString(json),
+          (err) => {
+            if (err) throw err;
+            console.log(`${id}.source.json has been saved!`);
+          }
+        );
+        return {
+          id,
+          json,
+        };
+      })
       .catch((e) => {
-        console.error("Failed to get xml for ", id, e);
+        console.error("Failed to get activity information for ", id, e);
       })
   );
 
-  // write activity source xml files
-  permitsXML.forEach(({ id, xml }) => {
-    fs.writeFile(path.join(outputDir, `${id}.xml`), xml, (err) => {
-      if (err) throw err;
-      console.log(`${id}.xml has been saved!`);
-    });
-  });
-
   Object.entries(topics).forEach(([id, permits]) => {
     const file = `${id}.json`;
     const data = {
       id,
-      permits: permits.map((permit) =>
-        sttrbuild(permitsXML.find((p) => p.id === permit.id).xml)
-      ),
+      permits: permits.map((permit) => {
+        const res = permitsConfig.find((p) => p.id === permit.id);
+        return {
+          version: res.json.version,
```

Comment:
Could there be a case where `res.json` or `res.json.version` is `undefined`?

Reply:
No there shouldn't be a reason for it to be undefined. If it does however the script should exit properly. Will double-check if that's the case. Thanks.